src/newProject/NewProjectPanel.ts

```diff
@@ -0,0 +1,356 @@
+      }
+    );
+    this.panel.webview.html = createNewProjectHtml(extensionPath);
+
+    this.panel.onDidDispose(() => this.dispose(), null);
+
+    this.panel.webview.onDidReceiveMessage((message) => {
+      switch (message.command) {
+        case "createProject":
+          if (
+            message.components &&
+            message.idf &&
+            message.template &&
+            message.tools &&
+            message.venv
+          ) {
+            this.createProject(
+              message.components,
+              message.idf,
+              message.template,
+              message.tools,
+              message.venv
+            );
+          }
+          break;
+        case "checkIsValid":
+          if (message.idf && message.tools && message.venv) {
+            this.checkSettings(message.idf, message.tools, message.venv);
+          }
+          break;
+        case "loadExamples":
+          if (message.idf_path) {
+            const examplesList = getExamplesList(message.idf_path);
+            this.panel.webview.postMessage({
+              command: "load_examples",
+              examples: examplesList,
+            });
+          }
+          break;
+        case "loadComponent":
+          vscode.window
+            .showOpenDialog({
+              canSelectFolders: true,
+              canSelectFiles: false,
+              canSelectMany: false,
+            })
+            .then((selectedFolder) => {
+              if (selectedFolder) {
+                const newComponent: IComponent = {
+                  name: basename(selectedFolder[0].fsPath),
+                  path: selectedFolder[0].fsPath,
+                };
+                this.panel.webview.postMessage({
+                  command: "component_list_add_path",
+                  new_component: newComponent,
+                });
+              }
+            });
+          break;
+        case "getIdfVersion":
+          if (message.idf_path) {
+            utils.getEspIdfVersion(message.idf_path.path).then((idfVersion) => {
+              this.panel.webview.postMessage({
+                command: "load_current_idf_version",
+                idfVersion,
+                idfPath: message.idf_path,
+              });
+              this.loadToolsVenvForIdf(
+                newProjectArgs.metadata,
+                message.idf_path
+              );
+            });
+          }
+        case "requestInitValues":
+          if (
+            newProjectArgs &&
+            newProjectArgs.metadata &&
+            newProjectArgs.metadata.idf &&
+            newProjectArgs.metadata.idf.length > 0 &&
+            newProjectArgs.metadata.venv &&
+            newProjectArgs.metadata.venv.length > 0 &&
+            newProjectArgs.metadata.tools &&
+            newProjectArgs.metadata.tools.length > 0
+          ) {
+            this.loadInitialMetadata(newProjectArgs.metadata);
+          } else {
+            Logger.infoNotify("No values available in metadata.json");
```

Comment:
For existing user (who are using the extension -- onboarding is done previously) its unable to find meradata file and throwing the error with blank webview

Reply:
Hmm good point, should we also include the current values of `idf.customExtraPaths`, `idf.pythonBinPath` and `idf.espIdfPath` to build an initial metadata.json if not available ?

Reply:
probably or ask the user to generate one himself or auto-generate by showing a popup like `Metadat file don't exists and do you like to auto generate or custom generate it`